### PR TITLE
feat(rulesets): add oas3_1-servers-in-webhook and oas3_1-callbacks-in…

### PR DESCRIPTION
**Pull request by kaylachun**
…-webhook rules

Fixes #[ISSUE_NUMBER].

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

> If indicated yes above, please describe the breaking change(s).
>
> **Remove this quote before creating the PR.**

**Screenshots**

If applicable, add screenshots or gifs to help demonstrate the changes. If not applicable, remove this screenshots section before creating the PR.

**Additional context**

Added two new spectral rules:
- oas3_1-servers-in-webhook
  - warns if servers is defined in a webhook
- oas3_1-callbacks-in-webhook
  - warns if callbacks is defined in webhook operation

### CODE REVIEW

File: packages/rulesets/src/oas/index.ts [hunk 1]
```diff
@@ -707,5 +707,23 @@ const ruleset = {
         },
       },
     },
+    'oas3_1-servers-in-webhook': {
+      message: 'Servers should not be defined in a webhook.',
+      formats: [oas3_1],
+      recommended: true,
+      given: ['$.webhooks.servers'],
```
Comment (daniel-white) [inline packages/rulesets/src/oas/index.ts]: i think we want this inside of the webhook operation, not at the webhooks object

File: packages/rulesets/src/oas/index.ts [hunk 2]
```diff
@@ -707,5 +707,23 @@ const ruleset = {
...
+        function: undefined,
+      },
+    },
+    'oas3_1-callbacks-in-webhook': {
+      message: 'Callbacks should not be defined in a webhook.',
+      formats: [oas3_1],
+      recommended: true,
+      given: ['$.webhooks[*][*].callbacks'],
+      then: {
+        function: undefined,
```
Comment (daniel-white) [inline packages/rulesets/src/oas/index.ts:734]: are these required?

Comment (kaylachun) [reply packages/rulesets/src/oas/index.ts:734]: Are you referring to the `function: undefined` part?  That function is checking if callbacks is undefined and if it's not then it'll show the warning.

File: packages/rulesets/src/oas/index.ts (hunk 1 above)
Comment (kaylachun) [reply packages/rulesets/src/oas/index.ts]: Ohhh okay servers can be defined on both.  Should the warning only be at the webhook operation level?

Comment (kaylachun) [reply packages/rulesets/src/oas/index.ts]: I'm reading through the oas docs and it seems like we should be showing a warning at both levels (the servers array at the path item object level would apply to all operations and servers at the operation level is for that operation only)

Comment (kaylachun) [reply packages/rulesets/src/oas/index.ts]: Ok added that here https://github.com/stoplightio/spectral/pull/2581/commits/fb14f54d522c3c6ac523dc59ab6c5149c327900d

**Review by mnaumanali94 — APPROVED** [review]

**Pull request merged**